x-pack/legacy/plugins/siem/public/containers/detection_engine/rules/use_create_packaged_rules.tsx

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useEffect, useState } from 'react';
+
+import { createPrepackagedRules } from './api';
+
+type Return = [boolean, boolean | null];
+
+interface UseCreatePackagedRules {
+  canUserCRUD: boolean | null;
+  hasIndexManage: boolean | null;
+  hasManageApiKey: boolean | null;
+  isAuthenticated: boolean | null;
+  isSignalIndexExists: boolean | null;
+}
+
+/**
+ * Hook for creating the packages rules
+ *
+ * @param hasIndexManage boolean
+ * @param hasManageApiKey boolean
+ * @param isAuthenticated boolean
+ * @param isSignalIndexExists boolean
+ *
+ */
```

Comment:
nit: Missing param ` * @param canUserCRUD boolean`. Also, we may want to start adding a `@returns [loading, hasCreatedPackageRules]` to our hooks as it's not clear _what_ is being returned exactly.

<img src="https://user-images.githubusercontent.com/2946766/72626805-988b2980-3908-11ea-9335-4cf40ad7a3f8.png" width=500>

<img src="https://user-images.githubusercontent.com/2946766/72626833-a2ad2800-3908-11ea-854d-243511bd7847.png" width=500>